Reference Models Centrally, set OpenAI Model to gpt-5-nano and Updated Spec Defaults

giterinhub opened the pull request:
## Feat: Add new conversation components and update model defaults

This pull request introduces documentation for two new conversation components (**GoogleAI** and **Ollama**) and updates the default LLM models for several existing components to newer versions.

A key change is that the default model for each conversation component can now be configured at runtime via a dedicated environment variable.

### Summary of Changes:

* **New Conversation Components:**
    * Added documentation for the `conversation.googleai` component, which defaults to the `gemini-2.5-flash-lite` model.
    * Added documentation for the `conversation.ollama` component, which defaults to the `llama3.2:latest` model.

* **Default Model Updates:**
    * **Anthropic:** The default model has been updated from `claude-3-5-sonnet-20240620` to `claude-sonnet-4-20250514`.
    * **Hugging Face:** The default model has been updated from `meta-llama/Meta-Llama-3-8B` to `deepseek-ai/DeepSeek-R1-Distill-Qwen-32B`.
    * **OpenAI:** The default model has been updated from `gpt-4-turbo` to `gpt-5-nano`.
    * **Mistral:** The default model remains `open-mistral-7b`.

* **Configuration via Environment Variables:**
    * Each conversation component's default model can now be overridden using a specific environment variable (e.g., `OPENAI_MODEL`, `GOOGLEAI_MODEL`, etc.).
    * The central environment variable reference page has been updated to include documentation for all new model environment variables:
        * `ANTHROPIC_MODEL`
        * `GOOGLEAI_MODEL`
        * `HUGGINGFACE_MODEL`
        * `MISTRAL_MODEL`
        * `OLLAMA_MODEL`
        * `OPENAI_MODEL`

msfussell commented:
@giterinhub - Is there a corresponding code issue tracking this that you are aware of?

giterinhub commented:
Created https://github.com/dapr/components-contrib/pull/3792

giterinhub commented:
> @giterinhub - Is there a corresponding code issue tracking this that you are aware of?

No issue I'm aware of, just created the PR. Thanks for the good time at the Boom Battle Mark!

giterinhub committed:
docs(conversation): reference env vars for default models in conversation components

Signed-off-by: Erin La <107987318+giterinhub@users.noreply.github.com>

giterinhub committed:
docs(conversation): add GoogleAI and Ollama conversation component docs

Signed-off-by: Erin La <107987318+giterinhub@users.noreply.github.com>

marcduiker committed:
Add kapa.ai script (#4853)

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

TechyMT committed:
fix: update Redis documentation to reference latest Sentinel documentation links (#4860)

Modified multiple Redis-related markdown files to update the links for the Sentinel master name documentation from the old URL to the latest version. This ensures users have access to the most current information regarding Redis Sentinel configuration.

Signed-off-by: Mustafa <mustafa.trunkwala@atlan.com>

TechyMT committed:
fix: update Redis documentation to reference latest Sentinel documentation links (#4860)

Modified multiple Redis-related markdown files to update the links for the Sentinel master name documentation from the old URL to the latest version. This ensures users have access to the most current information regarding Redis Sentinel configuration.

Signed-off-by: Mustafa <mustafa.trunkwala@atlan.com>

msfussell commented:
@giterinhub - Can you review this PR now the code is in. In particular can you address
@sicoyle @bibryam - Please review

1) How is the AZURE_OPENAI_MODEL env used and by which component? How does this relate to the apiType field here https://v1-16.docs.dapr.io/reference/components-reference/supported-conversation/openai/ used to set Azure usage?
2) Why have you included the OLLAMA model, there is an existing MD file, I presume this is not needed or update the existing one?
3) Can you update the https://v1-16.docs.dapr.io/reference/environment/ list with the list of these 7 env variables with a description of each one. 
4) Should there be an env var for AWS Bedrock and if not why?

giterinhub committed:
docs(conversation): align runtime model env vars & update component examples

- Add and document runtime env vars: OPENAI_MODEL, AZURE_OPENAI_MODEL, ANTHROPIC_MODEL, GOOGLEAI_MODEL, MISTRAL_MODEL, HUGGINGFACE_MODEL, OLLAMA_MODEL (with defaults)
- Update environment reference page with the new env var names and defaults
- Replace env-var placeholders in component YAML examples with literal default model names for OpenAI, GoogleAI, Anthropic, Mistral, Hugging Face and Ollama
- Add "(configurable via '<ENV>' environment variable)" to each provider's Spec metadata table to show override option
- Add Azure OpenAI usage section to `conversation.openai` doc (how to use `apiType: azure` and `AZURE_OPENAI_MODEL`)
- Clarify AWS Bedrock uses standard AWS auth (no Bedrock-specific env var added)
- Fix small markdown lint issues (trailing newlines/whitespace)

Signed-off-by: Erin La <107987318+giterinhub@users.noreply.github.com>

sicoyle commented on daprdocs/content/en/reference/environment/_index.md:
```diff
@@ -21,12 +21,20 @@ The following table lists the environment variables used by the Dapr runtime, CL
\ No newline at end of file
+| DAPR_HEALTH_TIMEOUT | SDKs | Sets the time on the "wait for sidecar" availability. Overrides the default timeout setting of 60 seconds. |
+| OPENAI_MODEL | Conversation components | Default model name used by the `conversation.openai` component at runtime when no `model` metadata is set in the component file. Default: `gpt-5-nano`. |
+| AZURE_OPENAI_MODEL | Conversation components / Azure config | Default Azure model name used by the `conversation.openai` component when `apiType: azure` is configured and no `model` metadata is provided. Default: `gpt-4.1-mini`. |
```
We use just OPENAI_MODEL for all openai compatible components so pls rm the AZURE_OPENAI_MODEL env var

giterinhub replied:
ok, but I use Azure OpenAI exclusively and kinda need this flexibility...

sicoyle commented on daprdocs/content/en/reference/components-reference/supported-conversation/openai.md:
```diff
@@ -34,9 +34,33 @@ The above example uses secrets as plain strings. It is recommended to use a secr
+
+The `conversation.openai` component can target either OpenAI's hosted API or Azure OpenAI. To select Azure OpenAI, set the component's `apiType` metadata to `azure` and provide the usual Azure-specific connection settings (for example, endpoint/region and API key) in the component configuration.
+
+When `apiType: azure` is used, the environment variable `AZURE_OPENAI_MODEL` may be set to provide a default Azure model identifier to use when the component's `model` metadata is not provided. This environment variable only affects the component when `apiType` is set to `azure` — the regular `DAPR_CONVERSATION_OPENAI_MODEL` remains the default for non-Azure OpenAI usage.
```
```suggestion
When `apiType: azure` is used, the environment variable `OPENAI_MODEL` may be set to provide a default Azure model identifier to use if the component's `model` metadata is not provided. In addition, the corresponding `endpoint` and `apiVersion` fields will be set to Azure specific settings.
```

sicoyle commented on daprdocs/content/en/reference/components-reference/supported-conversation/openai.md:
```diff
@@ -34,9 +34,33 @@ The above example uses secrets as plain strings. It is recommended to use a secr
+  - name: endpoint
+    value: "https://<your-resource-name>.openai.azure.com/"
+  - name: model
+    value: '${{AZURE_OPENAI_MODEL}}'
```
```suggestion
    value: 'gpt-4.1-nano'
```

sicoyle commented on daprdocs/content/en/reference/components-reference/supported-conversation/openai.md:
````diff
@@ -34,9 +34,33 @@ The above example uses secrets as plain strings. It is recommended to use a secr
+    value: '${{AZURE_OPENAI_MODEL}}'
+```
+
+If `model` is omitted from the component metadata and neither `AZURE_OPENAI_MODEL` nor `DAPR_CONVERSATION_OPENAI_MODEL` are set, the component falls back to its built-in default model.
````
```suggestion
If `model` is omitted from the component metadata and the model environment variable is not set, then the component falls back to its built-in default model.
```

sicoyle reviewed (commented):
few comments for this. Thank you for following up with the corresponding docs PR for your changes 🙌 🚀

giterinhub committed:
Update daprdocs/content/en/reference/components-reference/supported-conversation/openai.md

Co-authored-by: Sam <sam@diagrid.io>
Signed-off-by: Erin La <107987318+giterinhub@users.noreply.github.com>

giterinhub committed:
Update daprdocs/content/en/reference/components-reference/supported-conversation/openai.md

Co-authored-by: Sam <sam@diagrid.io>
Signed-off-by: Erin La <107987318+giterinhub@users.noreply.github.com>

giterinhub committed:
Update daprdocs/content/en/reference/components-reference/supported-conversation/openai.md

Co-authored-by: Sam <sam@diagrid.io>
Signed-off-by: Erin La <107987318+giterinhub@users.noreply.github.com>

giterinhub committed:
docs(environment): remove AZURE_OPENAI_MODEL variable from reference

Signed-off-by: GitHub <noreply@github.com>

giterinhub commented:
Thanks @sicoyle !

sicoyle reviewed (approved):
LGTM - thank you!

sicoyle commented:
@giterinhub any chance you can correct the DCO step on the PR? If you click in on the build step it has the cmds you can run to make it clean and green :)

marcduiker reviewed (approved):
LGTM

marcduiker committed:
Merge branch 'v1.15' into patch-1

msfussell commented:
Need to wait on this PR from main to branch https://github.com/dapr/components-contrib/pull/4029

marcduiker committed:
Prepare 1.15 for archival (#4857)

* Prepare 1.15 for archival

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

* Set archived_version to true in config.toml

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

---------

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

bibryam commented:
@msfussell 

> 
> 1. How is the AZURE_OPENAI_MODEL env used and by which component? How does this relate to the apiType field here https://v1-16.docs.dapr.io/reference/components-reference/supported-conversation/openai/ used to set Azure usage?


When apiType = "azure": Uses AZURE_OPENAI_MODEL environment variable (defaults to "gpt-4.1-nano")
When apiType is not "azure": Uses OPENAI_MODEL environment variable (defaults to "gpt-5-nano")


> 2. Why have you included the OLLAMA model, there is an existing MD file, I presume this is not needed or update the existing one?

I don't see additional MD file, maybe it was removed.

> 3. Can you update the https://v1-16.docs.dapr.io/reference/environment/ list with the list of these 7 env variables with a description of each one.
Seems that is done

> 4. Should there be an env var for AWS Bedrock and if not why?

IMO Bedrock should follow the same pattern and support env variable for model override.


Apart from these, this is a great PR (we need a similar PR for API KEY separately.

giterinhub commented:
Mentioned already that I like to have AZURE_OPENAI_MODEL and OPENAI_MODEL separately @bibryam I had to remove the entry for AZURE_OPENAI_MODEL as Sam wanted it gone ;)

sicoyle commented:
> Mentioned already that I like to have AZURE_OPENAI_MODEL and OPENAI_MODEL separately @bibryam I had to remove the entry for AZURE_OPENAI_MODEL as Sam wanted it gone ;)

AZURE_OPENAI_MODEL is not a thing... all components that use the conversation.openai type will just use the env var of OPENAI_MODEL if they choose to use the env var to set their model. There are several components, not just azure that are openai compatible, so we don't want a ton of env vars for each of them. Since they all can just use the openai component, then they can all use the openai env var. Examples include: Minimax, Qwen, Ernie, Azure, etc.

Editting to add a few clarifications:

There is logic under the hood that checks if apiType is set to azure. In that case, the appropriate Azure model is chosen instead of defaulting to the generic OpenAI model.

@bibryam Please create a separate issue if you’d like the API_KEY to also be set via an env var. That said, should all metadata fields be configurable this way? It feels like a lot—so where do we draw the line?

Looking back at the contrib PR, it appears we missed Bedrock. This shouldn’t block the current PR and can be added later, but for now, it will be missing.

giterinhub committed:
Merge branch 'v1.15' into patch-1

msfussell commented:
@giterinhub @sicoyle - This should now target v1.16 branch now or v1.17 if coming in next release.

sicoyle commented:
> @giterinhub @sicoyle - This should now target v1.16 branch now or v1.17 if coming in next release.

1.16 should be the docs target as it is in the 1.16 release branch.

sicoyle commented:
@giterinhub I updated the base branch as this should be based on v1.16, not v1.15. Can you resolve the conflicts pls, or even just checkout the v1.16 and cherry pick your commits from there if that's easier? 🙏
